esp32-mock-generator/src/main.cpp

```
@@ -0,0 +1,319 @@
+#include <WiFi.h>
+#include <PubSubClient.h>
+#include <ArduinoJson.h>
+#include "config.h"  // WiFi and MQTT credentials (gitignored)
+
+// ============================================================================
+// CONFIGURATION
+// ============================================================================
+// WiFi and MQTT broker settings are in include/config.h (gitignored)
+// Copy include/config.h.example to include/config.h and fill in your credentials
+
+const char* MQTT_TOPIC = "scada/measurements/node1";
```

Clang (14.0.6)

[warning] variable 'MQTT_TOPIC' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)